servers/republik/modules/crowdfundings/lib/scheduler/changeover.js

```diff
@@ -0,0 +1,146 @@
+const debug = require('debug')('crowdfundings:lib:scheduler:changeover')
+const moment = require('moment')
+const Promise = require('bluebird')
+
+const {
+  resolveMemberships,
+  findDormantMemberships
+} = require('../CustomPackages')
+
+const createCache = require('../cache')
+
+const changeover = async (args, { pgdb, mail: { enforceSubscriptions } }) => {
+  const endDate = moment()
+
+  const users = await pgdb.public.query(`
+    SELECT "userId" AS id FROM (
+      -- Active memberships and their MAX(membershipPeriods.endDate)
+      SELECT m.*, MAX(mp."endDate") AS "endDate"
+      FROM memberships m
+      INNER JOIN "membershipPeriods" mp
+        ON m.id = mp."membershipId"
+      INNER JOIN "membershipTypes" mt
+        ON m."membershipTypeId" = mt.id
+        AND mt.name != 'MONTHLY_ABO' -- Exclude subscription via Stripe
+      WHERE
+        m.active = true
+      GROUP BY 1
+    ) AS users
+    WHERE
+      -- Potential ending memberships to change over to dormant membership
+      ( "endDate" < :endDate )
+    GROUP BY 1
+  `, { endDate })
+
+  debug({
+    now: endDate.toDate(),
+    users: users.length
+  })
+
+  const stats = { changeover: 0 }
+
+  await Promise.each(
+    users,
+    async user => {
+      // Find all memberships a user owns
+      let userMemberships = await pgdb.public.memberships.find(
+        { userId: user.id }
+      )
+
+      userMemberships = await resolveMemberships({
+        memberships: userMemberships,
+        pgdb
+      })
+
+      if (userMemberships.filter(m => m.active).length > 1) {
+        console.error(
+          'changeover failed: multiple active memberships found',
+          { user }
+        )
+        return
+      }
+
+      const activeMembership = userMemberships.find(m => m.active)
```

Comment:
you could check if `activeMembership.periods.maxEndDate` is smaller than endDate. The SQL and code above should provide this, but it's all loosly coupled and it would make this more resilient to changes in the schema/business-logic.